apps/api/src/app/workflows-v2/usecases/build-variable-schema/build-available-variable-schema.usecase.ts

```diff
@@ -38,17 +38,22 @@ export class BuildAvailableVariableSchemaUsecase {
           additionalProperties: false,
         },
         steps: buildPreviousStepsSchema(previousSteps, workflow.payloadSchema),
-        payload: safePayloadSchema(workflow) || { type: 'object', description: 'Payload for the current step' },
+        payload: safePayloadSchema(workflow),
       },
       additionalProperties: false,
     } as const satisfies JSONSchemaDto;
   }
 }
-function safePayloadSchema(workflow: NotificationTemplateEntity): JSONSchemaDto | undefined {
+function safePayloadSchema(workflow: NotificationTemplateEntity): JSONSchemaDto {
   try {
-    return JSON.parse(workflow.payloadSchema);
+    // TODO: fix the type, we need to make sure we have the same data structure in the code
```

Comment:
we have inconsistency in the database where for framework workflows we have object and dashboard workflow, and we have it stringified to string